include/boost/random/detail/uniform_int_float.hpp

```diff
@@ -60,7 +60,7 @@ class uniform_int_float
     result_type operator()()
     {
         base_result range = static_cast<base_result>((max)())+1;
-        return static_cast<result_type>(_rng() * range);
+        return static_cast<result_type>(static_cast<base_result>(_rng() * range));
```

Comment:
> Did you push this update? I don't see it on my end.

My bad, now pushed.

 >There are several places that change to 3 space indent, when the 
surrounding code is all indented to 4 spaces.

Those should be fixed now.

 >traits.hpp has a *DETAIL* #include guard.
 >doc/Jamfile.v2 refers to detail/traits.hpp

Fixed.

 >In independent_bits_engine, I'd prefer all the private functions to be 
at the end.

Fixed.

 >base_result_type should be hidden from doxygen (and I'd also prefer 
not to rename it).

This one is trickier: base_result_type is used as the type used to seed 
by a single value: and since it just gets forwarded to the underlying 
generator, this is actually the correct type for a single valued seed.  
So the type *is* used in the interface, and is named to match the other 
public typedefs.  The alternative would be to make the seed type the 
same as the result type and explicitly cast to the base_result_type 
internally.  However, this has two key disadvantages - it's C++11 only, 
and it's also rather inefficient (we construct a wider/multiprecision 
type, just to narrow it again internally).  Oh, and it's illogical too ;)

BTW there are rather a lot of commits now, let me know if you want me to 
condense down to a single commit.

John.